### ISSUE TITLE
Client Certificate Issue TAXII Ingestor

### ISSUE THREAD

**Issue by 11eddie** [LuatixHQ/opencti#4534]
## Prerequisites

- [x] I read the [Deployment and Setup](https://www.notion.so/OpenCTI-Public-Knowledge-Base-d411e5e477734c59887dad3649f20518) section of the OpenCTI documentation as well as the [Troubleshooting](https://www.notion.so/Troubleshooting-ebc8fb04137d495aad917bc20340b9a6) page and didn't find anything relevant to my problem.
- [x] I went through old GitHub issues and couldn't find anything relevant
- [x] I googled the issue and didn't find anything relevant

## Description
I am trying to set up a taxii ingestor. I have entered my client certificate in pem format, the public key and the CA certificate below. They are all formatted correctly -----BEGINCERTIFICATE-----base64-----ENDCERTIFICATE----- no spaces or forbidden chars. 

Any idea why i'm getting the output below?

ERR [OPENCTI-MODULE] Taxii ingestion execution error for TAXII | category=APP error={"context":{},"message":"error:0480006C:PEM routines::no start line","name":"Error","stack":"Error: error:0480006C:PEM routines::no start line\n    at node:internal/tls/secure-context:70:13\n    at Array.forEach (<anonymous>)\n    at setCerts (node:internal/tls/secure-context:68:3)\n    at configSecureContext (node:internal/tls/secure-context:157:5)\n    at Object.createSecureContext (node:_tls_common:116:3)\n    at Object.connect (node:_tls_wrap:1748:48)\n    at Agent.createConnection (node:https:170:22)\n    at Agent.createSocket (node:_http_agent:342:26)\n    at Agent.addRequest (node:_http_agent:289:10)\n    at new ClientRequest (node:_http_client:337:16)\n    at Object.request (node:https:378:10)\n    at bb._performRequest (/opt/opencti/build/node_modules/follow-redirects/index.js:296:24)\n    at new RedirectableRequest (/opt/opencti/build/node_modules/follow-redirects/index.js:69:8)\n    at Object.request (/opt/opencti/build/node_modules/follow-redirects/index.js:535:14)\n    at asyncExecutor (/opt/opencti/build/node_modules/axios/lib/adapters/http.js:438:21)\n    at /opt/opencti/build/node_modules/axios/lib/adapters/http.js:143:5"} timestamp=2023-10-10T11:21:48.212Z version=5.10.3

## Environment
OpenCTI version=5.10.3
1. OS (where OpenCTI server runs): { Ubuntu}
2. OpenCTI version: { e.g. OpenCTI 1.0.2 }
3. OpenCTI client: { Docker}
4. Other environment details:

## Reproducible Steps
Using Client certificate Authentication in Taxii Ingestion

**Comment by zhanttt** [comment]
I experienced the same "no start line" error.
You need to base64 encode the entire pem file (including the BEGIN and END portions). Then take the b64 encoded result and convert it to a single line. I did the same for the crt file.
When I put these values in "cert" and "key", the error disappeared.

**Comment by 11eddie** [comment]
Thank you for the help, I hadn't realized I had to encode even the BEGIN and END portions 👍   

I updated OpenCTI and tried again following your advice but now I am getting a new error:

{"category":"APP","error":{"context":{},"message":"error:0480006C:PEM routines::no start line","name":"Error","stack":"Error: error:0480006C:PEM routines::no start line\n    at node:internal/tls/secure-context:70:13\n    at Array.forEach (<anonymous>)\n    at setCerts (node:internal/tls/secure-context:68:3)\n    at configSecureContext (node:internal/tls/secure-context:157:5)\n    at Object.createSecureContext (node:_tls_common:116:3)\n    at Object.connect (node:_tls_wrap:1748:48)\n    at Agent.createConnection (node:https:170:22)\n    at Agent.createSocket (node:_http_agent:342:26)\n    at Agent.addRequest (node:_http_agent:289:10)\n    at new ClientRequest (node:_http_client:337:16)\n    at Object.request (node:https:378:10)\n    at Zb._performRequest (/opt/opencti/build/node_modules/follow-redirects/index.js:296:24)\n    at new RedirectableRequest (/opt/opencti/build/node_modules/follow-redirects/index.js:69:8)\n    at Object.request (/opt/opencti/build/node_modules/follow-redirects/index.js:535:14)\n    at asyncExecutor (/opt/opencti/build/node_modules/axios/lib/adapters/http.js:438:21)\n    at /opt/opencti/build/node_modules/axios/lib/adapters/http.js:143:5"},"level":"error","message":"[OPENCTI-MODULE] Taxii ingestion execution error for TAXII","timestamp":"2023-11-03T18:01:06.389Z","version":"5.11.12"}

Does this mean that there's a /n somewhere in one of the certificates? 
I removed all spaces and encoded each with base64 in cyberchef so i'm not sure what my issue is now.

Also } keeps appearing at the end of the CA certificate for some reason

**Comment by zhanttt** [comment]
I'm guessing OpenCTI still doesn't like your cert and/or key values. There shouldn't be any spaces or newlines in the final string. Here's what I did on the terminal in Ubuntu:

1. `base64 my_key.key > my_key.b64`
2. `vim my_key.b54`
3. Make contents one line and don't put a space for the newlines: `:%j!`
4. Paste the result into OpenCTI and repeat for the cert file

For the "}" in CA cert, I submitted an issue: https://github.com/OpenCTI-Platform/opencti/issues/4796
